.github/workflows/pypi_publish.yml

```diff
@@ -1,37 +1,50 @@
-name: Python Packaging
+name: Packaging and Publishing
 
 on:
   push:
-
   workflow_dispatch:
+    inputs:
+      version:
+        description: "The version to publish, ie 1.0.0 or 1.0.0-dev1"
+        required: true
 
 jobs:
   build:
-    name: Build
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
+          ref: ${{ github.event_name == 'workflow_dispatch' && format('v{0}', github.event.inputs.version) || github.ref }}
+
       - uses: hynek/build-and-inspect-python-package@v2
 
+      - uses: actions/upload-artifact@v4
+        with:
+          name: Packages-${{ github.run_id }}
+          path: |
+            /tmp/baipp/dist/*.whl
```

Comment:
Had to specify this path for uploading build artifacts as the /dist directory was not matching the path in the previous step